sirepo/template/opal.py

```diff
@@ -222,6 +222,60 @@ def __init__(self):
         super().__init__(SIM_TYPE, self._FIELD_MAP)
 
     def to_madx(self, data):
+
+        def _get_len_by_id(data, id):
+            for e in data.models.elements:
+                if e._id == id:
+                    return e.l
```

Comment:
Need to handle the case where id is not found. Python implicitly returns None but if that is the desired behavior then be explicit and say `return None` (implicit behavior like this is a source of bugs https://lwn.net/Articles/794944/). Otherwise, add `raise AssertionError(f'id={id} not found in elements={data.models.elements}')`. This helps future readers know that you covered (or tried to) all cases.